### PR TITLE
22304-vmtarget-pharo-not-found-during-bootstrap

### PR DESCRIPTION
**Pull request by akgrant43**
vmtarget/pharo is referenced inconsistently resulting in the executable
not being found depending on the values of BOOTSTRAP_REPOSITORY and
BOOTSTRAP_CACHE.

- Move vmtarget to ${CACHE}
  This also simplifies tidying up afterwards as removing ${CACHE} will
  also remove vmtarget.
- Modify all references to vmtarget/pharo to use ${CACHE}

Fogbugz: https://pharo.fogbugz.com/f/cases/22304/vmtarget-pharo-not-found-during-bootstrap

### CODE REVIEW

**Comment by akgrant43** [comment]
CI test failures look unrelated - they're network errors after the build completes.

**Comment by akgrant43** [comment]
CI failures are again network related.

The latest commit (b20611f) isn't used during the build process, so couldn't have any affect.

**Comment by akgrant43** [comment]
CI failures are again network related.

File: bootstrap/scripts/snapshot.sh [hunk 1]
```diff
@@ -12,5 +12,5 @@ set -e
 . ${BOOTSTRAP_REPOSITORY:-.}/bootstrap/scripts/envvars.sh
 
 pushd "${BOOTSTRAP_REPOSITORY}"
-tar czf "${BOOTSTRAP_REPOSITORY}/snapshot.tar.gz" Pharo.image Pharo.changes pharo pharo-ui pharo-vm pharo-local vmtarget bootstrap-cache
+tar czf "${BOOTSTRAP_REPOSITORY}/snapshot.tar.gz" Pharo.image Pharo.changes pharo pharo-ui pharo-vm pharo-local ${CACHE}
```
Comment (pavel-krivanek) [inline bootstrap/scripts/snapshot.sh]: it should be BOOTSTRAP_CACHE here

Comment (akgrant43) [reply bootstrap/scripts/snapshot.sh]: Oops, sorry about that.

Actually snapshot.sh and restore.sh were both assuming that bootstrap.sh was being run from within the respository.  Now they should be run under the same conditions as bootstrap.sh.

**Comment by akgrant43** [comment]
The scripts are still making incorrect assumptions about their current working directory.

More changes to come (grrrr).

**Comment by akgrant43** [comment]
CI failures unrelated: same 3 ZnHTTPS tests

**Review by pavel-krivanek — APPROVED** [review]

**Comment by akgrant43** [comment]
Wow, that was fast!  Thanks Pavel & Marcus!

**Pull request merged**